adagucserverEC/utils/LayerMetadataStore.cpp

```diff
@@ -0,0 +1,298 @@
+  storeLayerMetadataStructIntoMetadataDb(myWMSLayer);
+  storeLayerDimensionListIntoMetadataDb(myWMSLayer);
+  storeLayerProjectionAndExtentListIntoMetadataDb(myWMSLayer);
+  storeLayerStyleListIntoMetadataDb(myWMSLayer);
+  return 0;
+}
+
+CT::string getLayerMetadataFromDb(MetadataLayer *myWMSLayer, CT::string metadataKey) {
+  CT::string layerName = myWMSLayer->dataSource->getLayerName();
+  CT::string datasetName = myWMSLayer->dataSource->srvParams->datasetLocation;
+  if (datasetName.empty()) {
+    // CDBDebug("Not a dataset");
+    return "";
+  }
+  return CDBFactory::getDBAdapter(myWMSLayer->dataSource->srvParams->cfg)->getLayerMetadata(datasetName, layerName, metadataKey);
+}
+
+int storeLayerMetadataInDb(MetadataLayer *myWMSLayer, CT::string metadataKey, std::string metadataBlob) {
+  try {
+    CT::string datasetName = myWMSLayer->dataSource->srvParams->datasetLocation;
+    if (datasetName.empty()) {
+      CDBDebug("Not a dataset");
+      return 1;
+    }
+    CT::string layerName = myWMSLayer->dataSource->getLayerName();
+
+    return CDBFactory::getDBAdapter(myWMSLayer->dataSource->srvParams->cfg)->storeLayerMetadata(datasetName, layerName, metadataKey, metadataBlob.c_str());
+  } catch (int e) {
+    return e;
+  }
+  return 0;
+}
+
+int storeLayerMetadataStructIntoMetadataDb(MetadataLayer *myWMSLayer) {
+  json layerMetadataItem;
+  layerMetadataItem["name"] = myWMSLayer->layerMetadata.name;
+  layerMetadataItem["title"] = myWMSLayer->layerMetadata.title;
+  layerMetadataItem["group"] = myWMSLayer->layerMetadata.group;
+  layerMetadataItem["abstract"] = myWMSLayer->layerMetadata.abstract;
+  layerMetadataItem["nativeepsg"] = myWMSLayer->layerMetadata.nativeEPSG;
+  layerMetadataItem["isqueryable"] = myWMSLayer->layerMetadata.isQueryable;
+  layerMetadataItem["latlonboxleft"] = myWMSLayer->layerMetadata.dfLatLonBBOX[0];
+  layerMetadataItem["latlonboxright"] = myWMSLayer->layerMetadata.dfLatLonBBOX[1];
+  layerMetadataItem["latlonboxbottom"] = myWMSLayer->layerMetadata.dfLatLonBBOX[2];
+  layerMetadataItem["latlonboxtop"] = myWMSLayer->layerMetadata.dfLatLonBBOX[3];
+  layerMetadataItem["bboxleft"] = myWMSLayer->layerMetadata.dfBBOX[0];
+  layerMetadataItem["bboxright"] = myWMSLayer->layerMetadata.dfBBOX[1];
+  layerMetadataItem["bboxbottom"] = myWMSLayer->layerMetadata.dfBBOX[2];
+  layerMetadataItem["bboxtop"] = myWMSLayer->layerMetadata.dfBBOX[3];
+  layerMetadataItem["width"] = myWMSLayer->layerMetadata.width;
+  layerMetadataItem["height"] = myWMSLayer->layerMetadata.height;
+  layerMetadataItem["cellsizex"] = myWMSLayer->layerMetadata.cellsizeX;
+  layerMetadataItem["cellsizey"] = myWMSLayer->layerMetadata.cellsizeY;
+  json variables;
+  for (auto lv : myWMSLayer->layerMetadata.variableList) {
+    json variable;
+    variable["units"] = lv->units;
+    variables.push_back(variable);
+  }
+  layerMetadataItem["variables"] = variables;
+
+  storeLayerMetadataInDb(myWMSLayer, "layermetadata", layerMetadataItem.dump());
```

Comment:
The other 3 store functions have `try catch`, but this one does not.

Reply:
storeLayerMetadataInDb does not throw anymore